src/db.c

```diff
@@ -322,12 +322,23 @@ kvobj *lookupKeyWriteOrReply(client *c, robj *key, robj *reply) {
  * link - Optional link to bucket where the key should be added.
  *          On return, get updated, by need, to the inserted key.
  */
-kvobj *dbAddByLink(redisDb *db, robj *key, robj **valref, dictEntryLink *link) {
+kvobj *dbAddByLink(redisDb *db, robj *key, robj **valref, dictEntryLink *link, long long expire) {
```

Comment:
Looking at the changes of this PR, I see many calls of `dbAddByLink` use `-1` and only one inside `dbAdd` that actually need this extra input variable. Maybe we should add a new function for that. WDYT?

Reply:
Tackled this issue. Pls review again.  Thanks.